crates/p2p/src/commands.rs

```diff
@@ -30,32 +39,50 @@ pub(crate) enum Command<DepositSetupPayload> {
     },
 }
 
-impl<DSP: Message> Command<DSP> {
-    /// Returns `gossipsub_msg::Body` with concatenated data for further signing.
-    pub(crate) fn into_gossipsub_msg(self) -> GossipsubMsgKind<DSP> {
-        match self {
-            Command::SendGenesisInfo {
+impl<DSP: Message + Clone> From<Command<DSP>> for GossipsubMsg<DSP> {
+    fn from(value: Command<DSP>) -> Self {
+        GossipsubMsg {
+            signature: value.signature,
+            key: value.key,
+            kind: value.kind.into(),
+        }
+    }
+}
+
+impl<DSP: Message + Default + Clone> CommandKind<DSP> {
+    pub fn sign_secp256k1(&self, keypair: &secp256k1::Keypair) -> Command<DSP> {
+        let kind: GossipsubMsgKind<DSP> = self.clone().into();
+        let msg = kind.content();
+        let signature = keypair.secret().sign(&msg);
+        Command {
+            key: keypair.public().clone().into(),
+            signature,
+            kind: self.clone(),
+        }
+    }
+}
+
+impl<DSP: Message + Clone> From<CommandKind<DSP>> for GossipsubMsgKind<DSP> {
+    fn from(value: CommandKind<DSP>) -> Self {
+        match value {
+            CommandKind::SendGenesisInfo {
```

Comment:
Also, that's why I didn't implement `From` before, as actually for `Command::RequestMessage` this method would require returning `Option<GossipsubMsgKind<DSP>>`